cupyx/jit/_compile.py

```diff
@@ -285,15 +297,22 @@ def _transpile_stmt(stmt, env):
         if not numpy.can_cast(
                 result.ctype.dtype, target.ctype.dtype, 'same_kind'):
             raise TypeError('dtype mismatch')
-        return f'{target.code} = {result.code};'
+        return [f'{target.code} = {result.code};']
     if isinstance(stmt, ast.For):
         raise NotImplementedError('Not implemented.')
     if isinstance(stmt, ast.AsyncFor):
         raise ValueError('`async for` is not allowed.')
     if isinstance(stmt, ast.While):
         raise NotImplementedError('Not implemented.')
     if isinstance(stmt, ast.If):
-        raise NotImplementedError('Not implemented.')
+        condition = _transpile_expr(stmt.test, env)
+        if is_constants([condition]):
+            stmts = stmt.body if condition.obj else stmt.orelse
+            return _transpile_stmts(stmts, env)
+        head = f'if ({condition.code})'
+        then_body = _transpile_stmts(stmt.body, env)
+        else_body = _transpile_stmts(stmt.orelse, env)
```

Comment:
How about an `if ... elif ... elif ... else` block? Does `elif` get handled here?

By the way, I noticed Cython translates such a block to a C switch when the condition is simple enough (ex: condition is a runtime constant). Do you think it's possible to do so here? Just curious 🙂

Reply:
Python's `ast` module handles `if ... elif ... else ...` equivalent to `if ... else { if ... else ... }`.
For example, in the test case [test_elif](https://github.com/cupy/cupy/pull/4646/files#diff-e0187cde046aeb239283aa518edbbac8aae3c09193ab03ccc82d2c607e79f1f3R324-R332), cupy.vectorize generates the following cuda code.
```cpp
template <typename T0, typename T1>
__device__ long long cupy_remainder_int64(T0 in0, T1 in1) {
    return (in0 - _floor_divide(in0, in1) * in1) * (in1 != 0);
}

__device__ long long func_if(long long x) {
  long long y;
  y = 0ll;
  if ((cupy_remainder_int64(x, 2ll) == 0ll)) {
    y = x;
  }
  else {
    if ((cupy_remainder_int64(x, 3ll) == 0ll)) {
      y = (-x);
    }
    else {
    }
  }
  return y;
}
```

Reply:
Cool, didn't know the ast's behavior. Thanks!